### PR TITLE
OD-9 [Feature] Integration DS Provider into LFD

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 
## Issue
OD-9 https://weboo.atlassian.net/jira/software/projects/OD/boards/9?selectedIssue=OD-9
## Description
Integration DS Provider into LFD
## Screenshots/screencasts
https://screencast-o-matic.com/watch/cY6lob7cuR
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -94,6 +95,43 @@ var DynamicLists = (function() {
     _this.init();
   }
 
+  function initDataSourceProvider(currentDataSourceId) {
+    var dataSourceData = {
+      dataSourceTitle: 'Profile data source',
+      dataSourceId: currentDataSourceId,
+      appId: Fliplet.Env.get('appId'),
+      default: {
+        name: 'LFD 0000',
```
Comment (tonytlwu) [inline js/interface.js:104]: This doesn't look like an appropriate name for the data source.

File: js/interface.js [hunk 2]
```diff
@@ -94,6 +95,43 @@ var DynamicLists = (function() {
     _this.init();
   }
 
+  function initDataSourceProvider(currentDataSourceId) {
+    var dataSourceData = {
+      dataSourceTitle: 'Profile data source',
```
Comment (tonytlwu) [inline js/interface.js:100]: The LFD doesn't need to select a **Profile data source**. This seems inappropriately labelled.

**Review by tonytlwu — CHANGES_REQUESTED** [review]
image

I see in the video (above) that it's still using the wrong data source selector dropdown. I'd like this verified with the latest data source provider UI to confirm the integration is working correctly.

**Comment by AndrRyaz** [comment]
@tonytlwu yes the video has an old version here is a new one 
https://screencast-o-matic.com/watch/cYXbliLic4

File: js/interface.js [hunk 3]
```diff
@@ -94,6 +95,43 @@ var DynamicLists = (function() {
     _this.init();
   }
 
+  function initDataSourceProvider(currentDataSourceId) {
+    var dataSourceData = {
+      dataSourceTitle: 'Your list data',
+      dataSourceId: currentDataSourceId,
+      appId: Fliplet.Env.get('appId'),
+      default: {
+        name: 'List from data source - ' + Fliplet.Env.get('appName'),
```
Comment (tonytlwu) [inline js/interface.js:104]: https://github.com/Fliplet/fliplet-widget-dynamic-lists/blob/4743c434ff975cece766f99c77487a02fb3725e6/js/interface.js#L1627

The current implementation gives you a default data source name based on a combination of the app name and the LFD layout.

We should keep the same functionality.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/interface.js (hunk 3 above)
Comment (AndrRyaz) [reply js/interface.js]: @tonytlwu  resolved, in the future I will use this structure

Comment (tonytlwu) [reply js/interface.js]: @AndrRyaz Essentially: pay attention to how features work before you make the changes, and don't break them.

**Review by tonytlwu — CHANGES_REQUESTED** [review]
0:34 in https://screencast-o-matic.com/watch/cYXbliLic4 shows that the column dropdown is empty after changing the data source, which appears like a bug. If not, there isn't a clear explanation for why it's OK for the dropdown to be empty.

**Comment by AndrRyaz** [comment]
@tonytlwu here is a new video https://screencast-o-matic.com/watch/cYXY6ALTKi, 
In the previous video, the selected date source did not have any fields in it - therefore it did not show anything, I already added default fields when creating the date source

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: High
Affected: Studio users configuring LFDs
Verify: All data source related features can be correctly configured to work as expected

Squash and merge: Integrate new data source pvovider

**Pull request merged**